navigator/runtime-experimental/src/main/java/com/freeletics/mad/navigator/compose/internal/StackEntry.kt

```diff
@@ -0,0 +1,25 @@
+package com.freeletics.mad.navigator.compose.internal
+
+import androidx.compose.runtime.Immutable
+import com.freeletics.mad.navigator.BaseRoute
+import com.freeletics.mad.navigator.NavRoute
+import com.freeletics.mad.navigator.compose.ContentDestination
+import com.freeletics.mad.navigator.internal.destinationId
+import dev.drewhamilton.poko.Poko
+
+@Poko
+@Immutable
+internal class StackEntry<T : BaseRoute>(
+    val id: Id,
+    val route: T,
+    val destination: ContentDestination<T>,
+){
+    val destinationId
+        get() = route.destinationId
+
+    val removable
+        get() = route is NavRoute
```

Comment:
wouldn't it be more explicit to check if `route !is NavRoot` instead?

Reply:
Changed it to a when [fc043a6](https://github.com/freeletics/mad/pull/369/commits/fc043a65903ad9fd31fb275aa3ee14603607ea13)